qudtlib-model/src/main/java/io/github/qudtlib/model/FactorUnits.java

```diff
@@ -564,24 +574,22 @@ private List<List<String>> permutate(List<String> strings) {
 
     private String getExponentString(int exponent) {
         int absExp = Math.abs(exponent);
-        switch (absExp) {
-            case 1:
-                return "";
-            case 2:
-                return "²";
-            case 3:
-                return "³";
-            case 4:
-                return "\u2074";
-            case 5:
-                return "\u2075";
-            case 6:
-                return "\u2076";
-            case 7:
-                return "\u2077";
-            default:
-                return "m";
-        }
+
+        if (absExp == 1) {
+            return "";
```

Comment:
although the solution seems more "elegant" i am not sure if we want to change the behaviour of this method. what do you say @fkleedorfer ?

Reply:
I made this better. Instead of chained replaces, it now turns the string of the number into a char stream, then does the mapping on char primitives and collects it in a StringBuilder. This should be a completely general O(log(n)) solution for positive integers translated to superscripts and avoids listing out all the superscript char literals.